src/node_errors.cc

```diff
@@ -843,59 +861,81 @@ void FatalException(Isolate* isolate,
     Abort();
   }
 
+  // Invoke process._fatalException() to give user a chance to handle it.
+  // We have to grab it from the process object since this has been
+  // monkey-patchable.
   Local<Object> process_object = env->process_object();
   Local<String> fatal_exception_string = env->fatal_exception_string();
   Local<Value> fatal_exception_function =
       process_object->Get(env->context(),
                           fatal_exception_string).ToLocalChecked();
-
+  // If the exception happens before process._fatalException is attached
+  // during bootstrap, or if the user has patched it incorrectly, just crash.
   if (!fatal_exception_function->IsFunction()) {
-    // Failed before the process._fatalException function was added!
-    // this is probably pretty bad.  Nothing to do but report and exit.
     ReportException(env, error, message);
     env->Exit(6);
-  } else {
-    errors::TryCatchScope fatal_try_catch(env);
-
-    // Do not call FatalException when _fatalException handler throws
-    fatal_try_catch.SetVerbose(false);
+    return;
+  }
 
+  MaybeLocal<Value> handled;
+  {
+    // We do not expect the global uncaught exception itself to throw any more
+    // exceptions. If it does, crash.
```

Comment:
Can you update this comment to reflect that `kFatal` does *not* crash the process, but rather calls `Environment::Exit`?

Reply:
Thanks for catching that. I'll update this to `exit the Node.js instance` as it may not be entirely clear what `Environment::Exit()` means (maybe `Node.js instance` is also vague, but I cannot think of a better term to describe "the main instance or the worker")

Reply:
I think `Node.js instance` is a pretty good term :)